From tonyj@suse.de Tue Mar  4 13:01:20 2008
From: Tony Jones <tonyj@suse.de>
Subject: memstick: convert struct class_device to struct device
Date: Fri, 22 Feb 2008 00:13:36 +0100
Message-Id: <1203635616.2702.8.camel@lov.site>
Mime-Version: 1.0
Content-Transfer-Encoding: 7bit
Status: RO
Content-Length: 348900

struct class_device is going away, struct device should be used instead.

Signed-off-by: Tony Jones <tonyj@suse.de>
Signed-off-by: Kay Sievers <kay.sievers@vrfy.org>
Cc: Alex Dubov <oakad@yahoo.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/memstick/core/memstick.c |   10 +++++-----
 include/linux/memstick.h         |    2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

--- a/drivers/memstick/core/memstick.c
+++ b/drivers/memstick/core/memstick.c
@@ -178,7 +178,7 @@ static struct bus_type memstick_bus_type
 	.resume         = memstick_device_resume
 };
 
-static void memstick_free(struct class_device *cdev)
+static void memstick_free(struct device *cdev)
 {
 	struct memstick_host *host = container_of(cdev, struct memstick_host,
 						  cdev);
@@ -489,7 +489,7 @@ struct memstick_host *memstick_alloc_hos
 		INIT_WORK(&host->media_checker, memstick_check);
 		host->cdev.class = &memstick_host_class;
 		host->cdev.dev = dev;
-		class_device_initialize(&host->cdev);
+		device_initialize(&host->cdev);
 	}
 	return host;
 }
@@ -515,7 +515,7 @@ int memstick_add_host(struct memstick_ho
 	snprintf(host->cdev.class_id, BUS_ID_SIZE,
 		 "memstick%u", host->id);
 
-	rc = class_device_add(&host->cdev);
+	rc = device_add(&host->cdev);
 	if (rc) {
 		spin_lock(&memstick_host_lock);
 		idr_remove(&memstick_host_idr, host->id);
@@ -546,7 +546,7 @@ void memstick_remove_host(struct memstic
 	spin_lock(&memstick_host_lock);
 	idr_remove(&memstick_host_idr, host->id);
 	spin_unlock(&memstick_host_lock);
-	class_device_del(&host->cdev);
+	device_del(&host->cdev);
 }
 EXPORT_SYMBOL(memstick_remove_host);
 
@@ -557,7 +557,7 @@ EXPORT_SYMBOL(memstick_remove_host);
 void memstick_free_host(struct memstick_host *host)
 {
 	mutex_destroy(&host->lock);
-	class_device_put(&host->cdev);
+	device_put(&host->cdev);
 }
 EXPORT_SYMBOL(memstick_free_host);
 
--- a/include/linux/memstick.h
+++ b/include/linux/memstick.h
@@ -235,7 +235,7 @@ struct memstick_host {
 #define MEMSTICK_CAP_AUTO_GET_INT  2
 
 	struct work_struct  media_checker;
-	struct class_device cdev;
+	struct device cdev;
 
 	struct memstick_dev *card;
 	unsigned int        retries;
